apps/examples/control_log/ctrlog.c

```diff
@@ -0,0 +1,391 @@
+		ETHARP_DEBUG,
+		NETIF_DEBUG,
+		PBUF_DEBUG,
+		API_LIB_DEBUG,
+		API_MSG_DEBUG,
+		SOCKETS_DEBUG,
+		ICMP_DEBUG,
+		IGMP_DEBUG,
+		INET_DEBUG,
+		IP_DEBUG,
+		IP_REASS_DEBUG,
+		RAW_DEBUG,
+		MEM_DEBUG,
+		MEMP_DEBUG,
+		SYS_DEBUG,
+		TIMERS_DEBUG,
+		TCP_DEBUG,
+		TCP_INPUT_DEBUG,
+		TCP_FR_DEBUG,
+		TCP_RTO_DEBUG,
+		TCP_CWND_DEBUG,
+		TCP_WND_DEBUG,
+		TCP_OUTPUT_DEBUG,
+		TCP_RST_DEBUG,
+		TCP_QLEN_DEBUG,
+		UDP_DEBUG,
+		TCPIP_DEBUG,
+		SLIP_DEBUG,
+		DHCP_DEBUG,
+		AUTOIP_DEBUG,
+		DNS_DEBUG,
+		IP6_DEBUG,
+		ND6_DEBUG,
+};
+
+static char *g_level_table_str[] = {
+		"error",
+		"info",
+		"verb",
+};
+
+static uint32_t g_level_table[] = {
+		NL_LEVEL_ERROR,
+		NL_LEVEL_INFO,
+		NL_LEVEL_VERB,
+};
+
+static char *g_color_table_str[] = {
+		"red", "yellow", "green", "default"};
+
+static nl_color_e g_color_table[] = {
+		NL_COLOR_RED, NL_COLOR_YELLOW, NL_COLOR_GREEN, NL_COLOR_GREEN};
+
+static char *g_mod_table_str[] = {
+		"wifi", "netlib", "lwip", "netmgr"};
+
+static netlog_module_e g_mod_table[] = {
+		NL_MOD_WIFI_MANAGER,
+		NL_MOD_NETLIB,
+		NL_MOD_LWIP,
+		NL_MOD_NET_MANAGER,
+};
+
+static char *g_opt_table_str[] = {
+		"on",
+		"off",
+};
+
+static nl_options g_opt_table[] = {
+		NL_OPT_ENABLE,
+		NL_OPT_DISABLE,
+};
+
+static uint32_t _get_lwip_mod(char *mod_str)
+{
+	printf("[pkbuild] %s \t%s:%d\n", mod_str, __FUNCTION__, __LINE__);
+	int size = sizeof(g_lwip_mod_table_str) / sizeof(char *);
+	for (int i = 0; i < size; i++) {
+		if ((strlen(mod_str) == strlen(g_lwip_mod_table_str[i]))
+        && strncmp(mod_str, g_lwip_mod_table_str[i], strlen(mod_str) + 1) == 0) {
+			return g_lwip_mod_table[i];
+		}
+	}
+	return UNKNOWN_DEBUG;
+}
+
+static uint32_t _get_level(char *level_str)
+{
+	printf("[pkbuild] level %s \t%s:%d\n", level_str, __FUNCTION__, __LINE__);
+	int size = sizeof(g_level_table_str) / sizeof(char *);
+	for (int i = 0; i < size; i++) {
+		if (strncmp(level_str, g_level_table_str[i], strlen(g_level_table_str[i]) + 1) == 0) {
+			printf("[pkbuild] i %d %d %s \t%s:%d\n", i, g_level_table[i], g_level_table_str[i], __FUNCTION__, __LINE__);
+			return g_level_table[i];
+		}
+	}
+	return NL_LEVEL_UNKNOWN;
+}
+
+static nl_color_e _get_color(char *color_str)
+{
+	printf("[pkbuild] %s \t%s:%d\n", color_str, __FUNCTION__, __LINE__);
+	int size = sizeof(g_color_table_str) / sizeof(char *);
+	for (int i = 0; i < size; i++) {
+		if (strncmp(color_str, g_color_table_str[i], strlen(g_color_table_str[i]) + 1) == 0) {
+			printf("[pkbuild] i %d %d %s \t%s:%d\n", i, g_color_table[i], g_color_table_str[i], __FUNCTION__, __LINE__);
+			return g_color_table[i];
+		}
+	}
+	return NL_COLOR_DEFAULT;
+}
+
+static netlog_module_e _get_mod(char *mod_str)
+{
+	printf("[pkbuild] %s \t%s:%d\n", mod_str, __FUNCTION__, __LINE__);
+	int size = sizeof(g_mod_table_str) / sizeof(char *);
+	for (int i = 0; i < size; i++) {
+		/* lwIP has additional submodule. So instead of comparing null
+     * chracter at the end of string, it won't compare it */
```

Comment:
Please change 'chracter' to 'character '

Reply:
Thannks I'll modify it